docs/design/2022-12-07-keyspace.md

```diff
@@ -0,0 +1,200 @@
+3. BR, Lightning, Dumpling, and CDC specify the keyspace by setting `keyspace-name` on the command line.
+
+## Implementation
+
+### PD
+
+#### PD Functionality
+
+PD provides an HTTP interface and an RPC interface to create, load, change the state, and configure keyspace metadata, and is responsible for interacting with the keyspace metadata store. Users can request the create keyspace interface by specifying the keyspace name, which generates the keyspace metadata.
+
+#### Keyspace Name
+
+**Keyspace Name** is specified by the creator during the creation of the keyspace, and once the keyspace is created, the keyspace name cannot be bound to another keyspace.
+
+#### Keyspace ID Allocation
+
+**Keyspace ID** is allocated by PD createKeyspace interface. The keyspace id is assigned by auto-incrementing the id, so keyspace ids are not re-assigned. The max keyspace id is 16777216, no new keyspaces can be created after the maximum keyspace id has been reached.
+
+#### Keyspace Metadata
+
+[**Keyspace Meta**](https://github.com/pingcap/kvproto/blob/d9297553c9009f569eaf4350f68a908f7811ee55/proto/keyspacepb.proto#L26C1-L33C2) is built when a keyspace is created, and the keyspace management interface operates on this metadata. The keyspace state in the metadata changes according to the keyspace state machine. The state change logic is as follows:
+![keyspace-state-machine.png](imgs%2Fkeyspace-state-machine.png)
+
+### TiDB
+
+#### Using API V2 and etcd namespace to isolate TiKV data and etcd path by Keyspace.
+
+1. Configure the keyspace name by setting `keyspace-name` in the configuration file. 
+   * When the TiDB server starts, it will open a TiKV driver. The driver has a TxnKV client object called `tikv.KVStore`. If `keyspace-name` is set, it will use `tikv.KVStore` via API V2 to access TiKV.
+2. The etcd path should have a different prefix based on the etcd namespace. 
+   * An etcd client is created during domain initialization. If `keyspace-name` is set, it will add an etcd namespace when creating the etcd client. 
+   * The format of the etcd namespace is `/keyspaces/tidb/$keyspaceId`.
+
+#### TiDB BR
+
+If "keyspace-name" is set when executing the BR command, it supports restore and backup for the specified keyspace data. The BR task will parse the old keyspace prefix from the source data, and get target keyspace from KVStore.codec. The old keyspace prefix and new keyspace prefix are used to update the old key prefix and new key prefix of rewrite rules. Restore checksum also requires the old and new keyspace prefixes. The backup task needs to add a target keyspace prefix when building ranges in `BackupRequest` to scan for the data of the specified keyspace in TiKV.
+
+#### TiDB Lightning
+
+If `keyspace-name` is set when executing the Lightning command, it will encode the key to add the keyspace prefix before appending rows to the SST file, and decode the key before recording conflict data into the TiDB table.
```

Comment:
```suggestion
If `keyspace-name` is set when executing the `Lightning` command, it will encode the key to add the keyspace prefix before appending rows to the SST file, and decode the key before recording conflict data into the TiDB table.
```